### PR TITLE
chore: Migrate to vite

### PR DESCRIPTION
**Pull request by eric-nguyen-cs**
### What
- Moving from Create React App to Vite for bundling, transpiling, building, ... the project: CRA is deprecated and has not been updated for over a year, the official React docs recommended using Vite to start a new project if you are not using a framework like Next.js or Remix.
- This should allow for easier integration with various librairies (like Tanstack Router or Shadcn/ui to name a few) and better help/support from the community when facing React tooling bugs

### CODE REVIEW

File: taxonomy-editor-frontend/README.md [hunk 1]
```diff
@@ -13,7 +13,7 @@ This is the main user interface developed in React, which works in conjunction w
 - [Material-UI](https://mui.com/)
 - [iso-639-1](https://www.npmjs.com/package/iso-639-1)
 
-This project was initially setup using [Create React App](https://github.com/facebook/create-react-app), [the documentation](https://create-react-app.dev/) might be useful.
+This project was initially setup using [Create React App](https://github.com/facebook/create-react-app), but was migrated to Vite. [Its documentation](https://vitejs.dev) might be useful.
```
Comment (alexgarel) [inline taxonomy-editor-frontend/README.md:16]: :+1:

**Review by alexgarel — COMMENTED** [review]
LGTM but I need to test.

**Comment by alexgarel** [comment]
@eric-nguyen-cs does not work for me. I think the container is trying to launch a browser at startup, while it should not inside docker… and it kills the node.

```
taxonomy_node-1         | 
taxonomy_node-1         | > taxonomy-editor-frontend@0.1.0 start
taxonomy_node-1         | > vite
taxonomy_node-1         | 
taxonomy_node-1         | 
taxonomy_node-1         |   VITE v5.1.3  ready in 204 ms
taxonomy_node-1         | 
taxonomy_node-1         |   ➜  Network: http://ui.taxonomy.localhost:8091/
taxonomy_node-1         | node:events:496
taxonomy_node-1         |       throw er; // Unhandled 'error' event
taxonomy_node-1         |       ^
taxonomy_node-1         | 
taxonomy_node-1         | Error: spawn xdg-open ENOENT
taxonomy_node-1         |     at ChildProcess._handle.onexit (node:internal/child_process:286:19)
taxonomy_node-1         |     at onErrorNT (node:internal/child_process:484:16)
taxonomy_node-1         |     at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
taxonomy_node-1         | Emitted 'error' event on ChildProcess instance at:
taxonomy_node-1         |     at ChildProcess._handle.onexit (node:internal/child_process:292:12)
taxonomy_node-1         |     at onErrorNT (node:internal/child_process:484:16)
taxonomy_node-1         |     at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
taxonomy_node-1         |   errno: -2,
taxonomy_node-1         |   code: 'ENOENT',
taxonomy_node-1         |   syscall: 'spawn xdg-open',
taxonomy_node-1         |   path: 'xdg-open',
taxonomy_node-1         |   spawnargs: [ 'http://ui.taxonomy.localhost:8091/' ]
taxonomy_node-1         | }
taxonomy_node-1         | 
taxonomy_node-1         | Node.js v20.11.1
taxonomy_node-1 exited with code 1
```

**Comment by eric-nguyen-cs** [comment]
> @eric-nguyen-cs does not work for me. I think the container is trying to launch a browser at startup, while it should not inside docker… and it kills the node.
> 
> ```
> taxonomy_node-1         | 
> taxonomy_node-1         | > taxonomy-editor-frontend@0.1.0 start
> taxonomy_node-1         | > vite
> taxonomy_node-1         | 
> taxonomy_node-1         | 
> taxonomy_node-1         |   VITE v5.1.3  ready in 204 ms
> taxonomy_node-1         | 
> taxonomy_node-1         |   ➜  Network: http://ui.taxonomy.localhost:8091/
> taxonomy_node-1         | node:events:496
> taxonomy_node-1         |       throw er; // Unhandled 'error' event
> taxonomy_node-1         |       ^
> taxonomy_node-1         | 
> taxonomy_node-1         | Error: spawn xdg-open ENOENT
> taxonomy_node-1         |     at ChildProcess._handle.onexit (node:internal/child_process:286:19)
> taxonomy_node-1         |     at onErrorNT (node:internal/child_process:484:16)
> taxonomy_node-1         |     at process.processTicksAndRejections (node:internal/process/task_queues:82:21)
> taxonomy_node-1         | Emitted 'error' event on ChildProcess instance at:
> taxonomy_node-1         |     at ChildProcess._handle.onexit (node:internal/child_process:292:12)
> taxonomy_node-1         |     at onErrorNT (node:internal/child_process:484:16)
> taxonomy_node-1         |     at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
> taxonomy_node-1         |   errno: -2,
> taxonomy_node-1         |   code: 'ENOENT',
> taxonomy_node-1         |   syscall: 'spawn xdg-open',
> taxonomy_node-1         |   path: 'xdg-open',
> taxonomy_node-1         |   spawnargs: [ 'http://ui.taxonomy.localhost:8091/' ]
> taxonomy_node-1         | }
> taxonomy_node-1         | 
> taxonomy_node-1         | Node.js v20.11.1
> taxonomy_node-1 exited with code 1
> ```

Weird, it seemed to work fine on my machine..
In any case, I've updated the code and the browser is not launched when starting the server

**Comment by alexgarel** [comment]
Still not working for me… I will try to debug.

**Review by alexgarel — CHANGES_REQUESTED** [review]
It does not work in static mode.

**Review by alexgarel — APPROVED** [review]
hum opening my PC and looking at it @eric-nguyen-cs I think my previous comment is a false alert, in the sense, it never worked as static in dev mode. I think we can try to deploy it tomorrow because it should work indeed.
The logic is in constants.ts and it seems ok to me.

**Pull request merged**